src/btree/row_modify.c

```diff
@@ -425,22 +422,11 @@ __wt_update_obsolete_check(
      */
     if (count > WT_THOUSAND) {
         WT_STAT_CONN_INCR(session, cache_eviction_force_long_update_list);
-        __wt_page_evict_soon(session, cbt->ref);
+        __wt_page_evict_soon(session, ref);
     }
 
     if (next != NULL)
         __wt_free_update_list(session, &next);
-    else {
-        /*
-         * If the list is long, don't retry checks on this page until the transaction state has
-         * moved forwards.
-         */
-        if (count > 20) {
```

Comment:
I thought this would be the version that was conditional on the new parameter. Am I missing something?

Reply:
I missed it to add back this change, Now I fixed all the locations.